users.js

```diff
@@ -0,0 +1,208 @@
+            "age": 17,
+            "address": "Simeon Veliki street 62, Troyan, Bulgaria"
+        },
+        {
+            "id": "1234567891",
+            "name": "mkoparanov",
+            "email": "mkoparanov@gmail.com",
+            "age": 21,
+            "address": "Simeon Veliki street 60, Troyan, Bulgaria"
+        },
+        {
+            "id": "1234567892",
+            "name": "vvalkov",
+            "email": "vvalkov@gmail.com",
+            "age": 24,
+            "address": "Vezhen street 28, Troyan, Bulgaria"
+        },
+        {
+            "id": "1234567893",
+            "name": "yvalkova",
+            "email": "yvalkova@gmail.com",
+            "age": 19,
+            "address": "Makedonia street 21, Troyan, Bulgaria"
+        },
+        {
+            "id": "1234567894",
+            "name": "svelikov",
+            "email": "svelikov@gmail.com",
+            "age": 20,
+            "address": "u tqh"
+        }
+    ]
+}`;
+
+//Using this object as a reference to the table rows.
+let trMap = {};
+
+const nameRegex = /[A-za-z]{1,30}/;
+const emailRegex = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+const addressRegex = /\w{1,300}/;
+
+if (!localStorage.getItem("database")) {
+  localStorage.setItem("database", database);
+}
+
+let tableBody = document
+  .getElementById("usersTable")
+  .getElementsByTagName("tbody")[0];
+
+const getUsers = async () => {
+  let usersJSON = localStorage.getItem("database") || database;
+  return JSON.parse(usersJSON).users;
+};
+
+const addUserToTable = (user) => {
+  let tableRow = document.createElement("tr");
+
+  let idTd = document.createElement("td");
+  let nameTd = document.createElement("td");
+  let emailTd = document.createElement("td");
+  let ageTd = document.createElement("td");
+  let addressTd = document.createElement("td");
+  let editTd = document.createElement("td");
+  let deleteTd = document.createElement("td");
+
+  let editButton = document.createElement("button");
+  let deleteButton = document.createElement("button");
+
+  editButton.innerHTML = "Edit";
+  editButton.className = "button";
+  editButton.addEventListener("click", () => loadUserToForm(user));
+
+  deleteButton.innerHTML = "Delete";
+  deleteButton.className = "button";
+  deleteButton.addEventListener("click", () => deleteUser(user));
+
+  editTd.appendChild(editButton);
+  deleteTd.appendChild(deleteButton);
+
+  idTd.innerHTML = user.id;
+  nameTd.innerHTML = user.name;
+  emailTd.innerHTML = user.email;
+  ageTd.innerHTML = user.age;
+  addressTd.innerHTML = user.address;
+
+  tableRow.appendChild(idTd);
+  tableRow.appendChild(nameTd);
+  tableRow.appendChild(emailTd);
+  tableRow.appendChild(ageTd);
+  tableRow.appendChild(addressTd);
+  tableRow.appendChild(editTd);
+  tableRow.appendChild(deleteTd);
+
+  tableBody.appendChild(tableRow);
+
+  trMap[user.id] = {
+    name: nameTd,
```

Comment:
So, you actually put the html elements in the model. Do you think that's a good idea?

Reply:
I made it in order not to loop trough the table after every update.

Reply:
I get it, but its not fine at all. There are problems with such approach and one of the biggest is that you couple the model with the presentation and even the persistence. If it happens that you have to change the table e.g. add or remove a column. If you already have users persisted, then you need to change the UI but also the model and the worst is that you would need to patch the DB and probably do some nasty hacks in the UI in order to make sure the data is properly visualized again.

Reply:
Yea I got it now. I will figure out a way to do it properly.